### PR TITLE
chore(components): setup detailed component doc pages

### PR DESCRIPTION
**Pull request by richbachman**
## Description of the change

This PR sets up the initial scaffolding for the new doc pages.
- Docs are now the entry point for each story
- Doc template is setup
- Anchor docs are done as an example

Screen Shot 2022-06-28 at 2 34 22 PM

This also updates Storybook and the add ons,  which shed light into some type errors and some things I'd like to adjust.

Most importantly, the `css` prop is now only on the Box primitive component. All other component styles/use-cases should be accounted for. I know there will be some changes, or situations that come up where a component style may need to be overridden. If thats needed, there are two approaches.

1. Use the standard `style` prop, and the theme tokens when possible.
2. Use the `styled` function and extend or override the original component.

I've added some comments in the threads below for examples.

Asana task: Partial of https://app.asana.com/0/1202513483755345/1202377854839749/f

## Testing the change

- [ ] Check out the new Anchor doc page

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] Issue from Asana has a link to this pull request

### CODE REVIEW

File: packages/components/src/components/anchor/src/types.ts [hunk 1]
```diff
@@ -8,14 +8,41 @@ export type AnchorSizes = "small" | "medium" | "large";
 
 export interface AnchorProps
   extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "css"> {
+  /**
+   * Sets the render element of the component. Prefer 'a' or 'button'.
+   */
```
Comment (RohitRajendran) [inline packages/components/src/components/anchor/src/types.ts:13]: Yaaas! Thanks for adding prop descriptions!

File: packages/components/src/components/anchor/src/types.ts [hunk 2]
```diff
@@ -8,14 +8,41 @@ export type AnchorSizes = "small" | "medium" | "large";
 
 export interface AnchorProps
   extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "css"> {
```
[the remark this answers is not in the document]
Comment (richbachman) [reply packages/components/src/components/anchor/src/types.ts:10]: I see both sides here. Using the HTML props from React, we don't have to do the work of adding those for each component. That said I do think there's a lot there that we don't really care about it/need. I looked at a few other open source design systems, and I see it done both ways. I don't have a strong opinion either way, so going to get the rest of the teams feedback on it. @bwll-tpio @jayarnielsen any thoughts?

[the remark this answers is not in the document]
Comment (jayarnielsen) [reply packages/components/src/components/anchor/src/types.ts:10]: I sort of like having access to any of the standard html attributes, but am also no super opinionated about it either.

File: packages/components/src/components/drag-and-drop-file-container/src/index.tsx [hunk 3]
```diff
@@ -96,14 +98,14 @@ const DragAndDropFileContainer = React.forwardRef<
               </Box>
             )}
 
-            <Text css={{ marginBottom: "$35" }}>
-              {isDragReject ? rejectText : uploadText}
-            </Text>
+            <Box css={{ marginBottom: "$35" }}>
+              <Text>{isDragReject ? rejectText : uploadText}</Text>
+            </Box>
```
Comment (richbachman) [inline packages/components/src/components/drag-and-drop-file-container/src/index.tsx:103]: The `css` should have only been on box, as its our utility component. Not really sure how this worked before.

File: packages/components/src/components/drag-and-drop-file-container/src/index.tsx [hunk 4]
```diff
@@ -96,14 +98,14 @@ const DragAndDropFileContainer = React.forwardRef<
...
 
             <Button
               variant="secondary"
               onClick={open}
-              css={
+              style={
                 isDragAccept
                   ? {
```
Comment (richbachman) [inline packages/components/src/components/drag-and-drop-file-container/src/index.tsx:110]: Similar here. We can always use `style` if we need to escape the system and override a component style.

File: packages/components/src/components/resource-link/src/ResourceAnchor.tsx [hunk 5]
```diff
@@ -1,47 +1,44 @@
...
-          "&:hover:after": {
-            opacity: "1",
-          },
-        }}
-        href={href}
-        noUnderline
-        {...props}
-        ref={ref}
-      >
+      <StyledAnchor href={href} noUnderline {...props} ref={ref}>
```
Comment (richbachman) [inline packages/components/src/components/resource-link/src/ResourceAnchor.tsx:39]: Or we can use the `styled` function to override/extend existing component styles.

File: package.json [hunk 6]
```diff
@@ -58,13 +58,13 @@
     "@commitlint/cli": "17.0.3",
     "@commitlint/config-conventional": "17.0.3",
     "@cspell/cspell-types": "6.1.2",
-    "@storybook/addon-actions": "6.4.21",
-    "@storybook/addon-docs": "6.4.21",
-    "@storybook/addon-essentials": "6.4.21",
-    "@storybook/addon-links": "6.4.21",
-    "@storybook/node-logger": "6.4.21",
-    "@storybook/preset-create-react-app": "4.1.0",
```
Comment (jayarnielsen) [inline package.json:66]: We aren't using this create react app preset are we?

Comment (richbachman) [reply package.json:66]: Nope. I'll kill it!

Comment (richbachman) [reply package.json:66]: We're not using the logger or actions either.

**Review by jayarnielsen — APPROVED** [review]
So happy that CSS weirdness is solved :raised_hands:

File: packages/components/src/components/anchor/src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (richbachman) [reply packages/components/src/components/anchor/src/types.ts]: @RohitRajendran I'll add this to our next FE sync, so we can chat more.

**Review by RohitRajendran — APPROVED** [review]
👍🏽

**Pull request merged**